variables.tf

```diff
@@ -93,6 +93,113 @@ variable "api_server_authorized_ip_ranges" {
   default     = null
 }
 
+variable "auto_scaler_profile_balance_similar_node_groups" {
+  description = "Enable or Disable the balance similar node groups. Defaults to false."
+  type        = bool
+  default     = false
+}
+
+variable "auto_scaler_profile_empty_bulk_delete_max" {
+  description = "The maximum number of empty nodes that can be deleted at the same time. Defaults to 10."
+  type        = number
+  default     = 10
+}
+
+variable "auto_scaler_profile_expander" {
+  description = "The expander for the node group. Possible values are `random`, `most-pods`, `least-waste`, `price`, `priority`. Defaults to `random`."
+  type        = string
+  default     = "random"
+  validation {
+    condition     = contains(["least-waste", "most-pods", "priority", "random"], var.auto_scaler_profile_expander)
+    error_message = "Must be either \"least-waste\", \"most-pods\", \"priority\" or \"random\"."
+  }
+}
+
+variable "auto_scaler_profile_max_graceful_termination_sec" {
+  description = "The maximum time in seconds for which a node can be gracefully terminated. Defaults to 600."
+  type        = number
+  default     = 600
+}
+
+variable "auto_scaler_profile_max_node_provisioning_time" {
+  description = "The maximum time in seconds for which cluster autoscaler waits for node to be provisioned. Defaults to 15m."
+  type        = string
+  default     = "15m"
+}
+
+variable "auto_scaler_profile_max_unready_nodes" {
+  description = "The maximum number of unready nodes that CA tolerates. Defaults to 3."
+  type        = number
+  default     = 3
+}
+
+variable "auto_scaler_profile_max_unready_percentage" {
+  description = "The maximum percentage of unready nodes in a node group. Defaults to 45."
+  type        = number
+  default     = 45
+}
+
+variable "auto_scaler_profile_new_pod_scale_up_delay" {
+  description = "The delay after pod starts, before it's added to calculation of pending pods. It should be used together with skip_nodes_with_local_storage. Defaults to 10s."
+  type        = string
+  default     = "10s"
+}
+
+variable "auto_scaler_profile_scale_down_delay_after_add" {
+  description = "The delay after node addition, before another node can be removed. Defaults to 10m."
+  type        = string
+  default     = "10m"
+}
+
+variable "auto_scaler_profile_scale_down_delay_after_delete" {
+  description = "The delay after node deletion, before another node can be removed. Defaults to 10s."
+  type        = string
+  default     = "10s"
```

Comment:
According to the [document](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/r/kubernetes_cluster.html.markdown?plain=1#L277):

>Defaults to the value used for `scan_interval`.

Which means if the user omit this value or set it to `null`, we should use `var.auto_scaler_profile_scan_interval` instead. I would suggest a new `local` in `locals.tf` file to implement this logic, and add a new unit test to ensure this logic. I would suggest you to read the `test/unit` folder to see how to write a unit test to do that, or you can just finish all other modifications then ping me, I could compose a new pr to your branch which contains the unit test.